### DISCUSSION TITLE
Timestamps and embeddings..?

### DISCUSSION THREAD

**Discussion by vstenby** [amazon-science/chronos-forecasting#446] [Q&A] [answered]
Hi - completely new to this, so sorry if this is a stupid question!

I want to use chronos-2 as an encoder for some time series.  

```python
from chronos import Chronos2Pipeline
import torch

# Load the model
pipeline = Chronos2Pipeline.from_pretrained("amazon/chronos-2", device_map="cuda")

# Give list of time series.
inputs = [
    torch.tensor([1.0, 2.0, 3.0, 4.0], dtype=torch.float32), 
] 

# Embed
emb, _ = pipeline.embed(inputs)
```

I notice that `predict_df` also takes `timestamp_column` as input when making predictions, but the `embed` function doesn't. Is there a way to parse timestamps or is that not needed? 

Thank you!

**Comment by vstenby** [comment]
Sorry for the bug label, couldn't add it with Q&A for some reason 😐

**Reply by abdulfatir** [reply]
No problem. Please use [discussions](https://github.com/amazon-science/chronos-forecasting/discussions) for Q/A.

**Reply by vstenby** [reply]
Sure, will do - thanks!

**Comment by abdulfatir** [comment, marked as answer]
@vstenby Chronos models actually do not use timestamps internally. The `predict_df` expects them to correctly format sequences as time series and to prevent incorrect use of the model (e.g., using on non-uniformly spaced time series). You should compare the `embed` method with `predict` or `predict_quantiles` which also does not take timestamps.

**Reply by vstenby** [reply]
Amazing, thanks @abdulfatir - that makes complete sense. Thanks for the quick reply!
